indexers/keyvalue/MongoDBIndexer/mongodbhandler.py

```diff
@@ -83,3 +87,30 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self.client.close()
         except pymongo.errors.PyMongoError as exp:
             raise MongoDBException(exp)
+
+    def delete(self, keys: Iterator[int], *args, **kwargs):
+        import pymongo
+        try:
+            count = self.collection.delete_many({'_id': {'$in': keys}}).deleted_count
+            self.logger.debug(f'deleted {count} documents in the database')
+        except pymongo.errors.PyMongoError as exp:
+            self.logger.error(f'got an error while deleting a document in the db {exp}')
+
+    def update(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
+        import pymongo
+        mongo_docs = [{'_id': i, 'values': j} for i, j in zip(keys, values)]
+        try:
+            # update_many updates several keys with the same op. / data.
+            # we need this instead
+            count = 0
+            for k, new_doc in zip(keys, mongo_docs):
+                inserted_doc = self.collection.find_one_and_replace(
```

Comment:
Replace sounds like MongoDB will overwrite the old content instead of updating it. Is that true? Then we first need to get the document from MongoDB, update it and only then replace the old content.

Reply:
What do you mean? Don't we just replace the document with the respective `id` with the values of the `new_doc`? 

For update we need to use mongo update operators, like these https://docs.mongodb.com/manual/reference/operator/update/#id1. We then need to specify the specific fields we're updating (if we use `$set`). But we don't get that sort of request from the `update` method, right? ` values: Iterator[bytes]` represents an entire new document value, not a more granular field update

Reply:
not for now but it will be included in the future

Reply:
Hm but is it then within the scope of this PR to handle that? Cause it will also require to establish how to read the `bytes` into JSON

Reply:
OK, sorry my misunderstanding, I thought update is only a partial update and not a full overwrite. This should be well documented, since it is non-obvious.

Reply:
I added a note to the docstring. I will also collect all these CRUD gotchas in the Jira ticket